tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,43 @@
+package checks
+
+import "github.com/tidwall/gjson"
+
+type CheckCode string
+
+type CheckError struct {
+	Text     string
+}
+
+type Check struct {
+	Code        CheckCode
+	Name        string
+	Description string
+	Check       func(*gjson.Result) []CheckError
+}
+
+type CheckCollection struct {
+	Name        string
+	Description string
+	Checks      []*Check
+}
+
+var CheckIntroducedEventExists = &Check{
```

Comment:
Probably worthwhile defining an interface for this:  

```golang
type Check interface {
  Name() string
  Description() string
  Check(record map[string]any)
}
```

Reply:
Done in 750a53d